### PR TITLE
Convert tmpdir test fixture to str after fsspec changes

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

Follow up to #7869
https://github.com/intake/filesystem_spec/pull/647 in fsspec was part of the 2021.06 release recently, which checks for

from https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture
> tmpdir is a py.path.local object which offers os.path methods and more.

This causes the instance check to fail. As a workaround, we convert the tmpdir to str until fsspec can accommodate this again

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/callbacks/test_early_stopping.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def test_resume_early_stopping_from_checkpoint(tmpdir):
     https://github.com/PyTorchLightning/pytorch-lightning/issues/1464
     https://github.com/PyTorchLightning/pytorch-lightning/issues/1463
     """
+    tmpdir = str(tmpdir)
```
Comment (tchaton) [inline tests/callbacks/test_early_stopping.py:59]: Can it be done in conftest.py directly ?

Comment (carmocca) [reply tests/callbacks/test_early_stopping.py:59]: Tangentially related, what if we converted all `tmpdir` inputs to `pathlib.Path` in `conftest`?

Comment (carmocca) [reply tests/callbacks/test_early_stopping.py:59]: it is a `py.path.LocalPath` not a `pathlib.Path`, there are small differences but the latter is much more common

Comment (carmocca) [reply tests/callbacks/test_early_stopping.py:59]: https://stackoverflow.com/a/67027860

**Pull request closed without merging**